### PR TITLE
Patel5pu code review

### PR DESCRIPTION
**Pull request by discospiff**
Analysis of the program.
Rentmider the app looked pretty good but to make the app look more impactful I add a separate section of gas as the Gas bills are usually a separate charge. Several components, including the DTO, as well as classes like Household, Member, and Payment, are presently absent from the codebase. Too many unneeded indents and line breaks hinder the project's readability.   Contrasting with Retrofit's focus on archiving facts, this app will only retain information entered by its users. Although the user interface is polished, there is need for improvement in its more technical front-end aspects, such as the lack of dollar sign formatting for the amount inputted. Also, when the user submits or saves their work, there should be a check to ensure they are entering text into the fields. Also, The code had unnecessary imports so removed few and added the necessary imports.

Was the program available in UC GitHub on time?
yes
Is the program documented/commented well enough for you to understand?
Yes
Does the program compile?
yes
Rationale behind your changes.

- As electric and gas bills come separately and we tend to log them on different time, I believe it should be put separately on different lines. 
- There were some imports that were not important, so I removed it. 
- I also decreased the lines of the code to make it compact.
- designed a new gas icon for the new section 

Things I learnt from code review
- I got know more about Ui Theme and icons 
- I now have a better idea of how to utilize jetpack compose to build out UI components using the Column and Row methods after examining their code.
- I learnt to create a section in the code
- I learnt to write the code more efficiently with using less lines

### CODE REVIEW

File: app/src/main/java/com/rentminder/MainActivity.kt [hunk 1]
```diff
@@ -41,17 +33,21 @@ class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
+
             RentMinderTheme {
                 Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colors.background
+                    modifier = Modifier.size(2800.dp).padding(end = 5.dp),
+                    color = MaterialTheme.colors.background,
```
Comment (discospiff) [inline app/src/main/java/com/rentminder/MainActivity.kt:40]: fillMaxSize() is more flexible than hardcoded layout.  Hardcoded dimensions won't make sense on all layouts.  Leave this as it were.

**Comment by discospiff** [comment]
It's not clear on how this branch reduces technical debt, and there's no discussion or source citations.  I'd orphan this branch.

File: app/src/main/java/com/rentminder/Icon&Text.kt [hunk 2]
```diff
@@ -1,22 +1,15 @@
...
-import androidx.compose.material.icons.outlined.Home
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import androidx.compose.ui.res.painterResource
-import androidx.compose.ui.text.TextStyle
-import androidx.compose.ui.text.style.TextDecoration
```
Comment (Pruthvi6797) [inline app/src/main/java/com/rentminder/Icon&Text.kt:19]: removed unnecessary imports

File: app/src/main/java/com/rentminder/Icon&Text.kt [hunk 3]
```diff
@@ -46,7 +39,23 @@ fun ElectricIconText() {
                 .size(30.dp)
         )
         Text(
-            text = "Electric/Gas",
```
Comment (Pruthvi6797) [inline app/src/main/java/com/rentminder/Icon&Text.kt:49]: Made electric a separate field

File: app/src/main/java/com/rentminder/MainActivity.kt [hunk 4]
```diff
@@ -132,6 +132,7 @@ fun TopToolBar() {
 fun EditBillAmounts() {
     var rentBill by remember { mutableStateOf("") }
     var electricBill by remember { mutableStateOf("") }
```
Comment (Pruthvi6797) [inline app/src/main/java/com/rentminder/MainActivity.kt:121]: Added a whole gas section

File: app/src/main/res/drawable/ic_gas_flame.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<vector android:height="24dp" android:tint="#000000"
```
Comment (Pruthvi6797) [inline app/src/main/res/drawable/ic_gas_flame.xml:1]: Added the icon details to the xml field

File: app/src/main/java/com/rentminder/MainActivity.kt (hunk 1 above)
Comment (Pruthvi6797) [reply app/src/main/java/com/rentminder/MainActivity.kt:40]: ok, I will change that, thank you

File: app/src/main/java/com/rentminder/MainActivity.kt [hunk 6]
```diff
@@ -194,9 +181,7 @@ fun EditBillAmounts() {
                             focusManager.clearFocus()
                         }
                     ),
-                    modifier = Modifier
-                        .width(100.dp)
-                        .height(55.dp),
+                    modifier = Modifier.width(100.dp).height(55.dp),
                     textStyle = TextStyle.Default.copy(fontSize = 18.sp)
```
Comment (Pruthvi6797) [inline app/src/main/java/com/rentminder/MainActivity.kt:185]: made the code more efficient.
https://kotlinlang.org/docs/coding-conventions.html#configure-style-in-ide

File: app/src/main/res/drawable/ic_gas_flame.xml (hunk 5 above)
Comment (Pruthvi6797) [reply app/src/main/res/drawable/ic_gas_flame.xml:1]: Added the XML file as it I needed to add the icon so had to put in details there

Comment (Pruthvi6797) [reply app/src/main/res/drawable/ic_gas_flame.xml:1]: https://google-developer-training.github.io/android-developer-fundamentals-course-concepts-v2/unit-2-user-experience/lesson-4-user-interaction/4-1-c-buttons-and-clickable-images/4-1-c-buttons-and-clickable-images.html

File: app/src/main/java/com/rentminder/Icon&Text.kt [hunk 7]
```diff
@@ -46,7 +39,23 @@ fun ElectricIconText() {
...
+    Row(verticalAlignment = Alignment.CenterVertically){
+        Icon(painterResource(
+            id = R.drawable.ic_gas_flame
+        ),
+            contentDescription = "gas flame Icon",
+            modifier = Modifier
+                .size(30.dp)
+        )
+        Text(
+            text = "Gas",
```
Comment (Pruthvi6797) [inline app/src/main/java/com/rentminder/Icon&Text.kt:58]: Added a gas section

File: app/src/main/java/com/rentminder/MainActivity.kt [hunk 8]
```diff
@@ -201,7 +202,32 @@ fun EditBillAmounts() {
...
+                        }
+                    ),
+                    modifier = Modifier
+                        .width(100.dp)
+                        .height(55.dp),
+                    textStyle = TextStyle.Default.copy(fontSize = 18.sp)
+                )
+                SaveRemindButton()
+            }
+        }
```
Comment (Pruthvi6797) [inline app/src/main/java/com/rentminder/MainActivity.kt:212]: Created a modifier for the gas section

File: app/src/main/res/drawable/ic_gas_flame.xml [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+    android:viewportHeight="24" android:viewportWidth="24"
+    android:width="24dp" xmlns:android="http://schemas.android.com/apk/res/android">
+    <path android:fillColor="@android:color/black" android:pathData="M12,2c-3.86,0 -7,3.14 -7,7c0,2.38 1.19,4.47 3,5.74L8,17c0,0.55 0.45,1 1,1h6c0.55,0 1,-0.45 1,-1v-2.26c1.81,-1.27 3,-3.36 3,-5.74c0,-3.86 -3.14,-7 -7,-7zM14.85,13.1l-0.85,0.6L14,16h-4v-2.3l-0.85,-0.6C7.8,12.16 7,10.63 7,9c0,-2.76 2.24,-5 5,-5s5,2.24 5,5c0,1.63 -0.8,3.16 -2.15,4.1z"/>
+</vector>
```
Comment (Pruthvi6797) [inline app/src/main/res/drawable/ic_gas_flame.xml:10]: Designed a new icon for gas section

File: app/src/main/java/com/rentminder/MainActivity.kt [hunk 10]
```diff
@@ -13,20 +11,14 @@ import androidx.compose.foundation.text.KeyboardOptions
...
-import androidx.compose.ui.Alignment.Companion.CenterHorizontally
 import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.LocalFocusManager
 import androidx.compose.ui.platform.LocalSoftwareKeyboardController
 import androidx.compose.ui.res.painterResource
-import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.TextStyle
```
Comment (Pruthvi6797) [inline app/src/main/java/com/rentminder/MainActivity.kt:22]: Removed unnecessary  imports

**Pull request closed without merging**